consensus/src/round_manager_test.rs

```diff
@@ -350,6 +354,75 @@ fn vote_on_successful_proposal() {
     });
 }
 
+#[test]
+/// In sync only mode, verify that the proposals are processed after we get out of the sync only mode.
+fn delay_proposal_processing_in_sync_only() {
+    let mut runtime = consensus_runtime();
+    let mut playground = NetworkPlayground::new(runtime.handle().clone());
+    // In order to observe the votes we're going to check proposal processing on the non-proposer
+    // node (which will send the votes to the proposer).
+    let mut nodes = NodeSetup::create_nodes(&mut playground, runtime.handle().clone(), 1);
+    let node = &mut nodes[0];
+
+    let genesis_qc = certificate_for_genesis();
+    timed_block_on(&mut runtime, async {
+        // Start round 1 and clear the message queue
+        node.next_proposal().await;
+
+        // Set sync only to true so that new proposal processing is delayed.
+        node.round_manager
+            .block_store
+            .set_back_pressure_for_test(true);
+        let proposal = Block::new_proposal(
+            Payload::empty(),
+            1,
+            1,
+            genesis_qc.clone(),
+            &node.signer,
+            Vec::new(),
+        );
+        let proposal_id = proposal.id();
+        node.round_manager
+            .process_proposal(proposal.clone())
+            .await
+            .unwrap();
+
+        // Wait for some time to ensure that the proposal was not processed
+        timeout(Duration::from_millis(200), node.next_vote())
+            .await
+            .unwrap_err();
+
+        node.round_manager
+            .process_verified_proposal(proposal.clone())
+            .await
+            .unwrap();
+
+        // Wait for some time to ensure that the proposal was not processed
+        timeout(Duration::from_millis(200), node.next_vote())
+            .await
+            .unwrap_err();
+
+        // Clear the sync only mode and process verified proposal and ensure it is processed now
+        node.round_manager
+            .block_store
+            .set_back_pressure_for_test(false);
+
+        node.round_manager
+            .process_verified_proposal(proposal)
```

Comment:
we should wait for the resend to trigger the vote instead of process it again?

Reply:
Yes, I thought about that, but unfortunately, the test is designed in a way that the round manager `start` loop is not executed, so events need to be sent manually by the test. Changing the test will be a lot of work and not sure if its worth it. Thoughts?